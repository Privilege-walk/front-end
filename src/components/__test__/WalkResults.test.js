import React from "react";
import { render, fireEvent, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import EventReults from '../../components/Walk/ResultsPage';
import eventResults from '../../api/mockData/event';
import questionAnswers from '../../api/mockData/questions';
import { Provider } from 'react-redux';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { test } from "@jest/globals";
var middlewares = [thunk]
var mockStore = configureMockStore(middlewares)


var mockFetchEventStats = jest.fn();
var mockFetchAnswerStats = jest.fn();

var mockUseParams = jest.fn().mockReturnValue({eventId:1, uniqueCode: "host"});

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn().mockReturnValue({eventId:1, uniqueCode: "host"}),
}));

afterEach(jest.clearAllMocks);


const setup = ({uniqueCode, eventId}) => {
    let utils;
    mockUseParams.mockReturnValue({uniqueCode, eventId});
    const store = mockStore({
        fetchEventStats: mockFetchEventStats,
        fetchAnswerStats: mockFetchAnswerStats
    });
    mockFetchEventStats.mockReturnValueOnce = {
        action: {
            payload: { data: eventResults}
        }
    }
    mockFetchAnswerStats.mockReturnValueOnce = {
        action: {
            payload: { data: questionAnswers}
        }
    }
    act (() => {
        utils = render(
        <Provider store={store}>
            <EventReults />
        </Provider>
        );
    })    
    return { ...utils };
}

test('render participant page results', () => {
    const params = { uniqueCode : "PH123", eventId: "1" }
    const utils = setup(params); 
    const { asFragment } = utils;
    expect(asFragment()).toMatchSnapshot();
});


test('render Host page results', () => {
    const params = { uniqueCode : "host", eventId: "1"}
    const utils = setup(params); 
    const { asFragment } = utils;
    expect(asFragment()).toMatchSnapshot();
});

