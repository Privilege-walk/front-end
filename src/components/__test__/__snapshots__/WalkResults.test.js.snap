// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Host page results 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiGrid-root css-1b6z6ql-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-19vs8dh-MuiGrid-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Perspective Walk
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      />
    </div>
    <div
      class="MuiGrid-root css-1xi1hvq-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-19vs8dh-MuiGrid-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Event Answer Statistics
        </div>
      </div>
      <div
        class="MuiGrid-root css-vj1n65-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-ece9u5"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="demo-simple-select-label"
            >
              Select Question
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="demo-simple-select"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="0"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1in441m"
                >
                  <span>
                    Select Question
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`render participant page results 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiGrid-root css-1b6z6ql-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-19vs8dh-MuiGrid-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Perspective Walk
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      />
    </div>
    <div
      class="MuiGrid-root css-1xi1hvq-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-19vs8dh-MuiGrid-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Event Answer Statistics
        </div>
      </div>
      <div
        class="MuiGrid-root css-vj1n65-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-ece9u5"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="demo-simple-select-label"
            >
              Select Question
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="demo-simple-select"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="0"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1in441m"
                >
                  <span>
                    Select Question
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      />
    </div>
  </div>
</DocumentFragment>
`;
